clinica/pipelines/t1_linear/t1_linear_utils.py

```diff
@@ -1,3 +1,32 @@
+from clinica.utils.exceptions import ClinicaParserError
+
+
+def extract_source_entities(ref_file):
+    import pathlib
+
+    # TODO: smarter entities extraction
+    source_entities = ("_").join(pathlib.Path(ref_file).stem.split("_")[:-1])
+    ext = "".join(pathlib.Path(ref_file).suffixes)
+
+    return source_entities, ext
+
+
+def construct_derivative_entities(pipeline_entities, pipeline_parameters):
```

Comment:
Could also be worth adding type hints for consistency (even though the other functions of this file do not seem to have them...).

Reply:
From what I remember, type hints did not couple well with Nipype Nodes - I can check again.